test/wallet-auction-test.js

```diff
@@ -0,0 +1,191 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+/* eslint no-return-assign: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const Chain = require('../lib/blockchain/chain');
+const WorkerPool = require('../lib/workers/workerpool');
+const Miner = require('../lib/mining/miner');
+const WalletDB = require('../lib/wallet/walletdb');
+const Network = require('../lib/protocol/network');
+const rules = require('../lib/covenants/rules');
+
+const network = Network.get('regtest');
+const NAME1 = rules.grindName(10, 20, network);
+
+const workers = new WorkerPool({
+  enabled: false
+});
+
+const chain = new Chain({
+  memory: true,
+  network,
+  workers
+});
+
+const miner = new Miner({
+  chain,
+  workers
+});
+
+const cpu = miner.cpu;
+
+const wdb = new WalletDB({
+  network: network,
+  workers: workers
+});
+
+describe('Wallet Auction', function() {
+  this.timeout(15000);
+
+  let winner;
+  const currentCBMaturity = network.coinbaseMaturity;
+
+  before(() => {
+    network.coinbaseMaturity = 1;
+  });
+
+  after(() => {
+    network.coinbaseMaturity = currentCBMaturity;
+  });
+
+  it('should open chain, miner and wallet', async () => {
+    await chain.open();
+    await miner.open();
+    await wdb.open();
+
+    winner = await wdb.create();
+
+    chain.on('connect', async (entry, block) => {
+      await wdb.addBlock(entry, block.txs);
+    });
+  });
+
+  it('should add addrs to miner', async () => {
+    const addr = await winner.createReceive();
+    miner.addresses = [addr.getAddress().toString(network)];
+  });
+
+  it('should mine 20 blocks', async () => {
```

Comment:
Is it possible to mine less blocks as to keep the test suite lighter? Could also move this out of an `it` and into `before`